src/Compilers/CSharp/Portable/Symbols/AliasSymbol.cs

```diff
@@ -99,10 +99,11 @@ internal static AliasSymbol CreateCustomDebugInfoAlias(NamespaceOrTypeSymbol tar
 
         internal AliasSymbol ToNewSubmission(CSharpCompilation compilation)
         {
-            Debug.Assert(_state.HasComplete(CompletionPart.AliasTarget));
             Debug.Assert(_binder.Compilation.IsSubmission);
 
-            var previousTarget = _aliasTarget;
+            // We can pass basesBeingResolved: null because base type cycles can't cross
+            // submission boundaries - there's no way to depend on a subsequent submission.
+            var previousTarget = GetAliasTarget(basesBeingResolved: null);
```

Comment:
@gafter I claim that I can pass null here even if we *are* in the middle of binding a base class, but I'd like to hear your thoughts.